### ISSUE TITLE
Navigation on Browse Mode: Add edit button

### ISSUE THREAD

**Issue by scruffian** [WordPress/gutenberg#50581]
Building on https://github.com/WordPress/gutenberg/issues/50578 and https://github.com/WordPress/gutenberg/issues/50579, we need to enable users to add pages to their navigation.

When users have a navigation selected (or if they have only one), add an edit button next the the "Navigation" title in the Navigation panel. This edit button will open the canvas preview to display the selected navigation in focus mode:

<img width="1512" alt="Screenshot 2023-05-12 at 12 40 32" src="https://github.com/WordPress/gutenberg/assets/275961/6a5b6fb8-a5ef-4a89-8ae4-af62a23b0e11">

**Comment by draganescu** [comment]
In the defunct navigation editor project we had a similar problem: how to edit a tree of blocks as in a way which is indicative that the tree is a navigation menu, without having a block that will contain the tree to style.

In the later stages of the project we landed on an UI which centered a rectangle on a gray canvas:

![image](https://github.com/WordPress/gutenberg/assets/107534/d8888132-d54c-4677-a942-17db051db1c2)

Should we try the same approach - which was attained by a lot of iteration on "horizontal" displays - to create the UX of editing a tree in a wider space where we also have inspectors for the containing blocks, but no inspector for the parent block? It seems fit for a data view of the navigation data.

Also in that screen:

- we used a wrapper navigation block only to allow the same rules of interaction but the block thre had its edit component filtered and most (all?) of the inspector options were removed.
- the block did allow renaming the navigation

![image](https://github.com/WordPress/gutenberg/assets/107534/9fb9b52c-b91c-49cd-aa2e-1b86048f5b77)

These bits from that investment should inform the work in #39286 which implements the bulk of this issue.

It's also worth noting that we have not solved the problem that since #34899 the navigation block inserts "navigation link" blocks via "direct insert" but the default state of the navigation link block is custom link. We would want that the default to be the page variation. This limitation will also show up in the isolated editor too.

Another thing to note from past lessons is that using blocks just for presetation (a fake navigation block to enable the menu building UX) leads to issues as we need to manage an "in-memory" entiti - see #34839

s
